### ISSUE TITLE
django_.django_dispatch fails when request type is ASGI -> request.environ not found. 

### ISSUE THREAD

**Issue by bliedblad** [soapteam/soapfish#85]
My Django application uses soapfish to provide a SOAP service. This service provides an async callback mechanism for long running functions for calls made on a remote SOAP server. 

I also need to provide feedback to the user when the long running process completes (without refresh). Websockets seemed to me the correct way to go, so I added Channels to my Django application. 

This caused the following line to choke since the ASGI request does not have an 'environ' variable:
`soap_request = SOAPRequest(DjangoEnvironWrapper(request.environ), body)`

I did a lot of Google searching, found a few notes suggesting I use the ASGI to WSGI adapter, but looks like Channels is so new that the adapter doesn't exist yet.

Long story short, found a post stating that `request.environ` is the same thing as `request.META`, which the ASGI request does have.

I have modified my local copy to read: 
`soap_request = SOAPRequest(DjangoEnvironWrapper(request.META), body)`

This appears to work, but am wondering if I'm going about this the wrong way and not sure if it warrants a Pull Request.

Thanks!

**Comment by FelixSchwarz** [comment]
Supporting ASGI directly sounds like a neat addition here. The main thing we need to make sure that the core library still works with plain WSGI and/or if the change requires a newer version of Django.
(For your pull request you might want to develop against the "soapfish" repo from FlightDataServices as I'll pull their changes soon enough. The pull request can go through this repo though.)

**Comment by FelixSchwarz** [comment]
just wanted to mention that I pulled a lot of changes from Nick so this repo should be suitable for pull requests again. :-)

**Comment by pope1ni** [comment]
We should look into this for soapfish 0.7.0 - support for channels in Django is still relatively new and under significant development and should not hold us back from getting 0.6.0 released.

Please feel free to submit a pull request.

**Comment by pope1ni** [comment]
We may well be interested in the future as we heavily use Django.

**Comment by bliedblad** [comment]
I'm probably not the best one to implement this change; however, should anyone need this functionality in the future... I believe the "fix" outlined in my initial comment works for both ASGI and WSGI. 

I no longer need this functionality as shortly after I discovered the work-around, I ran into a few roadblocks with Channels. 

After a LOT of searching I found another package called [Hendrix](https://github.com/hendrix/hendrix) which allows me to use websockets with WSGI (no workaround to soapfish required).

**Comment by ngnpope** [comment]
So I looked into this and found the following:
- `django.http.HttpRequest` has [`.META`](https://github.com/django/django/blob/d05ab13c56849ed09d02c1f188b7f47f0c090a2a/django/http/request.py#L57) only.
- `django.core.handlers.asgi.ASGIRequest` has [`.META`](https://github.com/django/django/blob/d05ab13c56849ed09d02c1f188b7f47f0c090a2a/django/core/handlers/asgi.py#L57) only.
- `django.core.handlers.wsgi.WSGIRequest` has [`.META`](https://github.com/django/django/blob/d05ab13c56849ed09d02c1f188b7f47f0c090a2a/django/core/handlers/wsgi.py#L77), and [`.environ`](https://github.com/django/django/blob/d05ab13c56849ed09d02c1f188b7f47f0c090a2a/django/core/handlers/wsgi.py#L70) which is an alias.
- `channels.http.AsgiRequest` has [`.META`](https://github.com/django/channels/blob/6af1bc3ab45f55e3f47d0d1d059d5db0a18a9581/channels/http.py#L63) only.

It's clear that it makes sense that this should be changed to `request.META` as proposed. Although it is possible that ASGI support would likely require further work, at least this issue can be resolved.
